fdbclient/BlobCipher.cpp

```diff
@@ -3107,6 +3107,142 @@ void testEncryptInplaceSingleAuthMode(const int minDomainId) {
 	TraceEvent("BlobCipherTestEncryptInplaceSingleAuthEnd").detail("Mode", authAlgoStr);
 }
 
+void testConfigurableEncryptionInvalidEncryptionKeyNoAuth(const int minDomainId) {
+	ASSERT(CLIENT_KNOBS->ENABLE_CONFIGURABLE_ENCRYPTION);
+
+	TraceEvent("TestConfigurableEncryptionInvalidEncryptKeyNoAuthStart");
+
+	Reference<BlobCipherKeyCache> cipherKeyCache = BlobCipherKeyCache::getInstance();
+
+	// Validate Encryption ops
+	Reference<BlobCipherKey> cipherKey = cipherKeyCache->getLatestCipherKey(minDomainId);
+	Reference<BlobCipherKey> headerCipherKey = cipherKeyCache->getLatestCipherKey(ENCRYPT_HEADER_DOMAIN_ID);
+	const int bufLen = deterministicRandom()->randomInt(786, 2127) + 512;
+	uint8_t orgData[bufLen];
+	deterministicRandom()->randomBytes(&orgData[0], bufLen);
+
+	Arena arena;
+	uint8_t iv[AES_256_IV_LENGTH];
+	deterministicRandom()->randomBytes(&iv[0], AES_256_IV_LENGTH);
+
+	EncryptBlobCipherAes265Ctr encryptor(cipherKey,
+	                                     headerCipherKey,
+	                                     iv,
+	                                     AES_256_IV_LENGTH,
+	                                     EncryptAuthTokenMode::ENCRYPT_HEADER_AUTH_TOKEN_MODE_NONE,
+	                                     BlobCipherMetrics::TEST);
+
+	BlobCipherEncryptHeaderRef headerRef;
+	StringRef encryptedBuf = encryptor.encrypt(&orgData[0], bufLen, &headerRef, arena);
+
+	// Test scenario where 'encryption key' with which the data was encrypted is 'different' from the one decryption
+	// gets attempted
+	AesCtrNoAuth noAuth = std::get<AesCtrNoAuth>(headerRef.algoHeader);
+	Reference<BlobCipherKey> tCipherKey = makeReference<BlobCipherKey>(cipherKey->getDomainId(),
+	                                                                   cipherKey->getBaseCipherId(),
+	                                                                   cipherKey->rawBaseCipher(),
+	                                                                   cipherKey->getBaseCipherLen(),
+	                                                                   cipherKey->getBaseCipherKCV(),
+	                                                                   cipherKey->getRefreshAtTS(),
+	                                                                   cipherKey->getExpireAtTS());
+	ASSERT(!tCipherKey->isEqual(cipherKey));
```

Comment:
Add a comment saying the cipher keys are not the same one because the salt is different? Took me a while to figure out..